contracts/utils/contracts/src/ReentrancyGuard.sol

```diff
@@ -21,25 +21,23 @@ pragma solidity ^0.5.5;
 
 contract ReentrancyGuard {
 
-    // Locked state of mutex
-    bool private locked = false;
+    // Mutex counter.
+    // Starts at 1 and increases whenever a nonReentrant function is called.
+    uint256 private reentrancyGuardCounter = 1;
 
-    /// @dev Functions with this modifer cannot be reentered. The mutex will be locked
-    ///      before function execution and unlocked after.
+    /// @dev Functions with this modifer cannot be reentered.
     modifier nonReentrant() {
-        // Ensure mutex is unlocked
+        // Increment the counter.
+        reentrancyGuardCounter += 1;
```

Comment:
We should be able to save 1 SLOAD here with:

```
uint256 localCounter = reentrancyGuardCounter + 1;
reentrancyGuardCounter = localCounter;
```

Reply:
Led me down a rabbit hole with remix and found `uint256 localCounter = ++reentrancyGuardCounter;` is actually the cheapest variant on 0.5.6.

Reply:
There may be a way to golf this down even further with some assembly :). But I feel like the savings would be pretty minimal.